src/engine/Eval_generic.ml

```diff
@@ -202,6 +204,31 @@ let rec eval env code =
   | G.Call ({ e = G.N (G.Id (("str", _), _)); _ }, (_, [ G.Arg e ], _)) ->
       let v = eval env e in
       eval_str env ~code v
+  (* Convert string to date *)
+  | G.Call ({ e = G.N (G.Id (("strptime", _), _)); _ }, (_, [ Arg e ], _)) -> (
+      let v = eval env e in
+      match v with
+      | String s -> (
+          let yyyy_mm_dd = String.split_on_char '-' s in
+          match yyyy_mm_dd with
+          | [ y; m; d ] -> (
+              match
+                (int_of_string_opt y, int_of_string_opt m, int_of_string_opt d)
+              with
+              | Some yv, Some mv, Some dv ->
+                  if CalendarLib.Date.is_valid_date yv mv dv then
```

Comment:
You can fold these together, if you use a `when` clause!
```
| Some yv, Some mv, Some dv when CalendarLib.Date.is_valid_date yv mv dv ->
```

Reply:
Saves a case :)

Reply:
Oo nice, didn't know this syntax, done!